v1.go

```diff
@@ -22,17 +23,20 @@ func initVersion1() *Header {
 }
 
 func parseVersion1(reader *bufio.Reader) (*Header, error) {
-	// Read until LF shows up, otherwise fail.
-	// At this point, can't be sure CR precedes LF which will be validated next.
-	line, err := reader.ReadString('\n')
-	if err != nil {
+	// The header must be terminated by CRLF and cannot be more than 107 bytes long.
+	buf, err := reader.Peek(107)
+	if err != nil && err != io.EOF {
 		return nil, ErrLineMustEndWithCrlf
```

Comment:
Please, add a new error here, something like `ErrLineSizeOverflow=errors.New("proxyproto: header is invalid, must be of size <= 108 bytes")`

Reply:
This is to be captured in the tests I asked for.

Reply:
ok, will do.